### PR TITLE
remove py2 remnants and use mocks in tests

### LINKED ISSUE #615: Drop support for 2.7
Since Python is officially dropping support for 2.7 on January 1st, 2020, I would like to drop support for Python 2 in jrnl 2.0 too. Happy to comb through the `v2.0-rc2` branch and make the adjustments, it'll make future maintenance a lot easier.

@wren, @MinchinWeb, @micahellison Any concerns or objections?

### LINKED ISSUE #667: Can't use encrypted journal on Windows 10 - TypeError: putwch() argument must be a unicode character, not int
In v2.0.0, when I try to make an encrypted journal in Windows 10, I get a Python error.

I defined a new encrypted journal "enc" in my jrnl.yaml:
``` yaml
journals:
  default: C:\Users\micah\.local\share\jrnl\journal.txt
  enc:
    journal: C:\Users\micah\.local\share\jrnl\enc.txt
    encrypt: true
```

Then I tried to write an entry to it from the shell:
```
C:\Users\micah>jrnl enc testing
Traceback (most recent call last):
  File "C:\Python37\Scripts\jrnl-script.py", line 11, in <module>
    load_entry_point('jrnl==2.0.0', 'console_scripts', 'jrnl')()
  File "c:\python37\lib\site-packages\jrnl\cli.py", line 210, in run
    journal = Journal.open_journal(journal_name, config)
  File "c:\python37\lib\site-packages\jrnl\Journal.py", line 359, in open_journal
    return EncryptedJournal.EncryptedJournal(name, **config).open()
  File "c:\python37\lib\site-packages\jrnl\EncryptedJournal.py", line 42, in open
    password = util.getpass("Enter password for new journal: ")
  File "c:\python37\lib\site-packages\jrnl\util.py", line 52, in getpass
    return gp.getpass(bytes(prompt))
  File "c:\python37\lib\getpass.py", line 103, in win_getpass
    msvcrt.putwch(c)
TypeError: putwch() argument must be a unicode character, not int
```

### LINKED ISSUE #700: Prompts display b''
## **I'm submitting a ...**
  - [x] bug report
  - [ ] feature request
  - [ ] support request

## Regarding bug reports:

* **Please tell us about your environment:** 
  - Jrnl version: (run `jrnl -v`):  `jrnl version v2.0.1`
  - How you installed Jrnl: `pipx install jrnl`
	
  - Operating system [MacOS, Linux, Windows?]: macOS Catalina

* **What is the current behavior?**

Prompts display `b''`

* **Please provide the steps to reproduce and if possible a minimal demo of the problem** 

Open a password-encrypted journal

* **What is the expected behavior?**

Prompt displays without `b''`

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

https://github.com/jrnl-org/jrnl/blob/53677a9df144ebb45bd7e8f0541a6abc38574c4d/jrnl/util.py#L121-L125

### LINKED ISSUE #701: Remove Python 2 compatibility code.
## **I'm submitting a ...**
  - [ ] bug report
  - [ ] feature request
  - [x] support request

While poking around in the code, I noticed some Python 2-compatibility code in https://github.com/jrnl-org/jrnl/blob/master/jrnl/util.py . Now that jrnl only supports Python 3.7, this code can be removed. 

I also recommend running [pyupgrade](https://github.com/asottile/pyupgrade).

### PR DESCRIPTION
**Pull request by pspeter**
This PR should completely remove all the Python 2 compatibility features and other leftovers.
Closes #701, closes #615, fixes #700, might fix #667.

I also removed the wonky testing utilities in jrnl.utils and instead used unittest.mock.patch to simulate user inputs in the tests. It's a bit unusual as well this way, but at least we don't have to mess with replacing std.stdin and sys.stderr and what not. If somebody knows a simpler way to simulate user input that works with builtins.input, getpass.getpass as well as sys.stdin.read, let me know.

This is a bit of a bigger PR, but the tests all pass.

## Todo
- [x] Remove from \_\_future\_\_ imports
- [x] Switch util functions to builtins print()/input() 
- [x] Rework tests to mock user input
- [x] Remove all mentions of unicode
- [x] Change str.format() to f-strings
- [x] Change codecs.open to builtins.open

### Checklist
- [x] The code change is tested and works locally.
- [x] Tests pass. Your PR cannot be merged unless tests pass
- [x] There is no commented out code in this PR.
- [x] Have you followed the guidelines in our Contributing document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](../pulls) for the same update/change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?

### CODE REVIEW

File: features/upgrade.feature [hunk 1]
```diff
@@ -13,11 +13,11 @@ Feature: Upgrading Journals from 1.x.x to 2.x.x
 
     Scenario: Upgrading a journal encrypted with jrnl 1.x
         Given we use the config "encrypted_old.json"
-        When we run "jrnl -n 1" and enter
```
Comment (wren) [inline features/upgrade.feature]: 🥇

File: jrnl/Journal.py [hunk 2]
```diff
@@ -118,7 +115,7 @@ def _parse(self, journal_txt):
         # Initialise our current entry
         entries = []
 
-        date_blob_re = re.compile("(?:^|\n)\[([^\\]]+)\] ")
+        date_blob_re = re.compile("(?:^|\n)\\[([^\\]]+)\\] ")
```
Comment (wren) [inline jrnl/Journal.py]: (optional nitpick): use raw string here for regex to avoid double-escaping

File: jrnl/cli.py [hunk 3]
```diff
@@ -175,12 +171,12 @@ def run(manual_args=None):
...
             args.limit = int(args.text[0].lstrip("-"))
             args.text = args.text[1:]
-        except:
+        except ValueError:
             pass
 
     log.debug('Using journal "%s"', journal_name)
     mode_compose, mode_export, mode_import = guess_mode(args, config)
-
+
```
Comment (wren) [inline jrnl/cli.py]: trailing whitespace :(

Comment (wren) [reply jrnl/cli.py]: also, this is optional and a nitpick

File: jrnl/plugins/yaml_exporter.py [hunk 4]
```diff
@@ -18,7 +17,8 @@ class YAMLExporter(TextExporter):
     def export_entry(cls, entry, to_multifile=True):
         """Returns a markdown representation of a single entry, with YAML front matter."""
         if to_multifile is False:
-            print("{}ERROR{}: YAML export must be to individual files. Please specify a directory to export to.".format("\033[31m", "\033[0m", file=sys.stderr))
+            print("{}ERROR{}: YAML export must be to individual files. "
```
Comment (wren) [inline jrnl/plugins/yaml_exporter.py]: (optional) missed `ERROR_COLOR`, etc here

File: jrnl/upgrade.py [hunk 5]
```diff
@@ -19,14 +18,14 @@ def backup(filename, binary=False):
...
-    with codecs.open(config_path, "r", "utf-8") as f:
+    with open(config_path, "r", encoding="utf-8") as f:
         config_file = f.read()
     if not config_file.strip().startswith("{"):
         return
 
     config = util.load_config(config_path)
 
-    util.prompt("""Welcome to jrnl {}.
+    print("""Welcome to jrnl {}.
```
Comment (wren) [inline jrnl/upgrade.py]: This is no longer going to stderr.

**Comment by pspeter** [comment]
Glad to help! But I could've fixed the nitpicks before merging (and squash the fixup commits) if you waited a little bit ;).  
I'll rebase my other PRs now.

**Pull request merged**